From 7c082f15c981b134fc52325853d319ff394dcf6e Mon Sep 17 00:00:00 2001
From: "tao.zeng" <tao.zeng@amlogic.com>
Date: Wed, 19 Feb 2014 15:43:09 +0800
Subject: [PATCH 3539/5965] PD #83628: PMU driver update for AML1216

1. change config for HeadPhone detect and wifi power config;
2. limit min charge current to 750mA and fix wrong config for usb voltage limit;
---
 arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd |  3 ++-
 drivers/amlogic/power/aml_pmu/aml1216-sply.c | 12 ++++++------
 2 files changed, 8 insertions(+), 7 deletions(-)

diff --git a/arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd b/arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd
index d6829071f4de..ca791b76c309 100755
--- a/arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd
+++ b/arch/arm/boot/dts/amlogic/meson8_k150_v1.dtd
@@ -1396,7 +1396,7 @@ lcd_CLAA101FP05XG:lcd_CLAA101FP05XG{	//k101
         I2S_SCLK = "GPIOAO_9";
         I2S_LRCLK = "GPIOAO_10";
         I2S_ODAT = "GPIOAO_11";
-        hp_paraments = <800 300 0 5 1>;
+        hp_paraments = <500 300 0 5 1>;
 		pinctrl-names = "aml_snd_m8";
 		pinctrl-0 = <&audio_pins>;
 		status = "okay"; 
@@ -1630,6 +1630,7 @@ lcd_CLAA101FP05XG:lcd_CLAA101FP05XG{	//k101
             dev_name = "wifi_power";
             status = "okay";
             power_gpio = "GPIOAO_6";
+            power_gpio2 = "GPIOX_11";
 	};  
     
 /// ***************************************************************************************
diff --git a/drivers/amlogic/power/aml_pmu/aml1216-sply.c b/drivers/amlogic/power/aml_pmu/aml1216-sply.c
index f0007ff755f4..d1f5955ab83e 100755
--- a/drivers/amlogic/power/aml_pmu/aml1216-sply.c
+++ b/drivers/amlogic/power/aml_pmu/aml1216-sply.c
@@ -254,19 +254,19 @@ int aml1216_set_usb_voltage_limit(int voltage)
     val &= ~(0xc0);
     switch (voltage) {
     case 4300:
-        val |= (0x01 << 6);
+        val |= (0x01 << 5);
         break;
 
     case 4400:
-        val |= (0x02 << 6);
+        val |= (0x02 << 5);
         break;
 
     case 4500:
-        val |= (0x00 << 6);
+        val |= (0x00 << 5);
         break;
 
     case 4600:
-        val |= (0x03 << 6);
+        val |= (0x03 << 5);
         break;
     
     default:
@@ -304,8 +304,8 @@ int aml1216_set_charging_current(int curr)
     } else {                                    // input is charge ratio
         curr = (curr * aml1216_battery->pmu_battery_cap) / 100 + 100; 
     } 
-    if (curr > 1050) {                       // limit current to 1050mA 
-        curr = 1050;    
+    if (curr < 750) {                       // limit current to 600mA for stable issue
+        curr = 750;    
     }
     idx_to = (curr - 300) / 150;
     aml1216_read(0x012b, (unsigned char *)&val);
-- 
2.19.0

